app/main.py

```diff
@@ -1,13 +1,33 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
+    return {
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8"
+    }
     pass
 
-
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
+    return {"errors":
+        [{format_linter_error(error)} for error in errors],
+    "path": file_path,
+    "status": "passed"
+    if errors == []
+    else "failed"
+    }
     pass
 
-
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
+    return [
+            {"errors": [], "path": list(linter_report)[0], "status": "passed"},
+            {
+                "errors": [
+                    {format_single_linter_file(linter_report["errors"])}
+                    for errors in linter_report],
+                "path": list(linter_report)[1],
+                "status": "passed"
+                if "errors" == []
```

Comment:
Here you are comparing a string 'errors' with an empty list. This condition will never be true. It seems like you wanted to check if there are any errors in your report. You should adjust your condition to correctly check this.